### PR TITLE
Track relevant product events

### LINKED ISSUE #11156: Shopping: Telemetry
Tracking for now:

Editor:

* Product insertion
* Errors when fetching products

Settings:

* Shopping provider change
* API connection test (click on button + errors)

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

This way we can gain some insights into how the feature is used

## Summary

Adds events tracking for various user actions related to shopping, e.g. insertion.

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

Now tracking product insertion, search, configuration changes, and errors.

### Does this PR have a legal-related impact?

No

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11156

### CODE REVIEW

File: packages/story-editor/src/components/library/panes/shopping/shoppingPane.js
Comment (timarney) [inline packages/story-editor/src/components/library/panes/shopping/shoppingPane.js]: Think we need to get rid of the = '' here ya? otherwise we don't end up using the search term

Comment (swissspidy) [reply packages/story-editor/src/components/library/panes/shopping/shoppingPane.js]: Ah yes! Copy/paste mistake

File: packages/story-editor/src/components/library/panes/shopping/shoppingPane.js
Comment (timarney) [inline packages/story-editor/src/components/library/panes/shopping/shoppingPane.js:123]: Not sure if this is the spot that's causing a bug but, if you clear the search term the products query doesn't re-run

Comment (swissspidy) [reply packages/story-editor/src/components/library/panes/shopping/shoppingPane.js:123]: Ah good catch, fixed now in 8a15c36a58a13faa144167f67f258ce700ef7b4c

File: packages/story-editor/src/components/library/panes/shopping/shoppingPane.js
Comment (spacedmonkey) [inline packages/story-editor/src/components/library/panes/shopping/shoppingPane.js:175]: In my pagination PR, I merge the two uses of useStory.

**Review by timarney — APPROVED** [review]

**Review by spacedmonkey — CHANGES_REQUESTED** [review]
I would recommend waiting until this is merged - https://github.com/GoogleForCreators/web-stories-wp/pull/11683

**Pull request merged**
